lib/screens/addCourse.dart

```diff
@@ -0,0 +1,199 @@
+
+  String error = '';
+  bool loading = false;
+
+  // text field state
+  var name = TextEditingController();
+  var abbreviation = TextEditingController();
+
+  // dropdown button state
+  Year year;
+
+  String makeAbbreviation(String courseName) {
+    String temp = courseName;
+    if (courseName.contains("]")) {
+      temp = courseName.substring(courseName.indexOf("]") + 1);
+    }
+    if (temp == "") return "";
+    List<String> parts = temp.replaceAll("and", "&").split(" ");
+    DateTime now = DateTime.now();
+    String result =
+        (now.month >= 6 ? "[F" : "[S") + (now.year % 100).toString() + "]";
+    parts.forEach((part) {
+      result += part == "" ? "" : part[0].toUpperCase();
+    });
+    return result;
+  }
+
+  @override
+  void initState() {
+    name = TextEditingController();
+    abbreviation = TextEditingController();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return loading
+        ? Loading()
+        : Scaffold(
+            resizeToAvoidBottomPadding: false,
+            backgroundColor: Colors.white,
+            appBar: AppBar(
+                backgroundColor: ColorsStyle.primary,
+                elevation: 0.0,
+                title: Text('Adding Course')),
+            body: Container(
+              padding: EdgeInsets.symmetric(vertical: 20.0, horizontal: 50.0),
+              child: Form(
+                key: _formKey,
+                child: Column(
+                  children: <Widget>[
+                    Row(
+                      mainAxisAlignment: MainAxisAlignment.start,
+                      children: [
+                        Text("Name"),
+                      ],
+                    ),
+                    TextFormField(
+                      controller: name,
+                      decoration:
+                          textInputDecoration.copyWith(hintText: 'name'),
+                      validator: (val) =>
+                          val.isEmpty ? 'Enter course\'s name' : null,
+                      onChanged: (val) {
+                        setState(() {
+                          abbreviation.text = makeAbbreviation(val);
+                        });
+                      },
+                    ),
+                    SizedBox(height: 20.0),
+                    Row(
+                      mainAxisAlignment: MainAxisAlignment.start,
+                      children: [
+                        Text("Abbreviation"),
+                      ],
+                    ),
+                    TextFormField(
+                      controller: abbreviation,
+                      decoration: textInputDecoration.copyWith(
+                          hintText: 'abbreviation'),
+                      validator: (val) => val.isEmpty ||
+                              !(val.startsWith("[F") || val.startsWith("[S"))
+                          ? 'Enter abbreviation in format: \'[S20] ABC\''
+                          : null,
```

Comment:
why? if the value is empty or doesn't start with "[F" or "[S" it won't, but your example will validate. I haven't extra time for coming up with a good enough algorithm for abbreviation validation, I agree that it is dumbfuckery, but if you have any better ideas let me know, we will change it))))0) I think for the start it is enough to avoid simple format violation

Reply:
I don't think we should force the user to type this on his own🤔 At least []-part could be referenced by dropdown. But to be honest I think I just don't like the idea of writing it and believe that it should be fetched from somewhere, like moodle. Maybe it is good to mock this data for now and say that "in the future, there would be a connection with moodle".

Reply:
We hope that admins would not be braindead and they could check auto-generated abbreviation and edit if needed